### ISSUE TITLE
`/rest/v1.2/read/following/mine`: Fix return type of `total_subscriptions` to be a number

### ISSUE THREAD

**Issue by mashikag** [Automattic/wp-calypso#75911]
![Image](https://user-images.githubusercontent.com/2019970/232835101-5be3480d-b72b-4447-83fb-73e7347e4679.png)

As per the screenshot above, the current return value is string. The expected return value for `total_subscriptions` is a number.

**Comment by mashikag** [comment]
Resolved by: D108479-code

**Comment by ivan-ottinger** [comment]
This also fixes an issue where in the case of no subscriptions, the following message doesn't show up:

![Image](https://user-images.githubusercontent.com/25105483/236417140-98599ad4-3a8f-4197-ad3e-f1b1746b32c6.png)

That's because the notice check expects `totalCount` (which is derived from `total_subscriptions`) to be number as well:

```jsx
	if ( ! isLoading && ! totalCount ) {
		return <Notice type="warning">{ translate( 'You are not subscribed to any sites.' ) }</Notice>;
	}
```
